fix(ui): add checkboxes when enabling multi-selection mode in FileExplorer

ef-molina opened the pull request:
### ✨ Description

What does this PR change?  
Adds checkboxes when enabling multi-select mode.

Users can:
- **Shift-click** to select a contiguous range of files using an anchor
- **Ctrl/Cmd-click** to toggle individual files
- Select files by clicking **anywhere on the row** or directly on the checkbox

Once two or more files are selected, the UI enters multi-select mode:
- Checkboxes become visible
- Modifier keys are no longer required to continue selecting or deselecting files
- Right-click preserves the current selection and reflects the correct count

The behavior mirrors common desktop file explorers and VS Code’s file tree UX.

### 🔗 Related Issue

Closes #8423 

### 🎨 Frontend Checklist

- [x] Code builds without errors (`npm run build`)
- [ ] All existing E2E tests pass (`npm run test:e2e`)
- [x] Screenshots or video recordings attached showing the `UI` changes

https://github.com/user-attachments/assets/aca192f1-7dc9-42b2-ba57-dee277966f54

### 📝 Additional Notes

This implementation was inspired by PR #11974, which was closed due to divergence, but followed the same UX direction.

Also, Range selection is currently computed over a flattened tree structure.

As a result:
- Shift-selecting **downward** from a collapsed folder includes its descendants in the selection
- Shift-selecting **upward** from a collapsed folder selects only the folder row itself
- Descendants included via downward selection become visible as selected once the folder is expanded

ef-molina committed:
fix(ui): add checkboxes when enabling multi-selection mode in FileExplorer

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

elevatebart commented on ui/src/components/inputs/FileExplorer.vue:
```diff
@@ -107,14 +107,14 @@
             :load="filesStore.loadNodes"
             :data="filesStore.fileTree"
-            highlightCurrent
+            :highlightCurrent="false"
```
Since hilightCurrent defaults to false you can simply omit it

https://element-plus.org/en-US/component/tree#attributes

```suggestion
```

elevatebart commented on ui/src/components/inputs/FileExplorer.vue:
```diff
@@ -107,14 +107,14 @@
             nodeKey="id"
             v-loading="filesStore.fileTree === undefined"
-            :props="{class: nodeClass, isLeaf: 'leaf'}"
+            :props="{class: nodeClass, isLeaf: 'leaf', key: 'id'}"
```
What does this change do ? I do not see anything about a key in the `props` attribut of `<el-tree>`.
https://element-plus.org/en-US/component/tree#props

If it does nothing, can you remove it please?

ef-molina replied:
Yes i will remove it i thought i needed this at first, but did not end up using it.

elevatebart commented on ui/src/components/inputs/FileExplorer.vue:
```diff
@@ -499,42 +521,140 @@
+    function handleNodeClick(data: any, node: ElTreeNode, event: MouseEvent | null = null, forceToggle = false) {
         const path = filesStore.getPath(node.data.id) ?? "";
-        const flatList = flattenTree(filesStore.fileTree);
+        const flatList = flatTree.value;
```
Thank you !!! ❤️

elevatebart commented on ui/src/components/inputs/FileExplorer.vue:
```diff
@@ -929,11 +1036,13 @@
 
         .node {
-            --el-tree-node-content-height: fit-content;
+            /* --el-tree-node-content-height: fit-content; */
```
When you decide code should not be used, please remove it.
Outdated comments tend to create visual noise. They make relevant code harder to find.

```suggestion
```

elevatebart reviewed (changes requested):
Great work!! 🎉 thank you much.

I have one UX ask though.
When a users starts to see checkboxes, they want to be able to exit from it quickly.
To turn off the checkboxes and unselect all other files, Users expect to click on the name of a file. Then everything should deselect.

Do you think it's possible?

There are also a few nitpicks, easy fixes.

Thank you again!

ef-molina commented:
Thanks a lot for the feedback — that makes sense.

I can update the behavior so that when multi-select mode is active, a normal click on a file row (without Shift/Ctrl and outside the checkbox) will:

1. clear the current selection

2. exit multi-select mode (hide checkboxes)

3. select only the clicked file

4. open it as usual

Please let me know if I'm misunderstanding the desired behavior. 
I will work on this soon. 

note: I accidentally requested a review.

elevatebart reviewed (changes requested):
So you can tell me when you are ready to get re-reviewed

ef-molina committed:
fix(ui): exit multi-select on row click in FileExplorer

ef-molina commented:
Hi @elevatebart,

I’ve addressed the requested UX change so that a normal file click exits multi-select mode and opens the file, and cleaned up the noted nits (removed unused props, comments, and params).

Let me know if you’d like any further tweaks.

elevatebart committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

ef-molina commented:
@elevatebart — quick FYI: I noticed the failing Storybook interaction in CI and verified it locally (InputsForm → InputTypes); it passes without changes on my end.

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

MilosPaunovic committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

elevatebart committed:
Merge branch 'develop' into 8423-add-checkboxes-to-namespace-files

elevatebart reviewed (approved):
very good PR thanks for the work. Some nitpixks that can be handled later
